pkg/kubeconfig/service_accounts.go

```diff
@@ -52,7 +52,7 @@ func (s *ServiceAccounts) Find(accessLocation AccessLocation, saName string) (Ac
 
 	pgoForSA := AccessInfo{
 		Name:       accessLocation.Name,
-		Namespace:  "", // Assume kubeconfig contains preferred namespace from SA
```

Comment:
👋 It's been a while since I looked through this code. And at first glance I am wondering if accessLocation.Namespace is what we are wanting here? Or does that refer to a different namespace again?

Reply:
We have [this](https://github.com/carvel-dev/kapp-controller/blob/786f84b0cc75b135423cf7f7010fb7fbe681f628/pkg/deploy/kapp.go#L257-L259) check, so `-n installationNamespace` would be added if user provides spec.installationNamesapce. We could also add the namespace directly to the kubeconfig, but using AccessInfo seemed inline to the kapp behaviour, that by default ns from kubeconfig is used, and we can override that via `-n` flag.